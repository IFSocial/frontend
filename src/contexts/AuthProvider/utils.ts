import api from '../../services/api';
import { IUser } from './types';

// Salva o usuario no BD local
export function setUserLocalStorage(user: IUser | null) {
  localStorage.setItem('u', JSON.stringify(user));
}

// Checa se existe o usuario no BD local
export function getUserLocalStorage() {
  const json = localStorage.getItem('u');
  if (!json) {
    return null;
  }

  const user = JSON.parse(json);

  return user ?? null;
}

// Faz request do login passando email e senha como payload
export async function LoginRequest(email: string, password: string) {
  try {
    const request = await api.post('/users/signin', { email, password });

    return request.data;
  } catch {
    return null;
  }
}

// Faz request do cadastrp passando nome, email e senha como payload
export async function SignupRequest(
  name: string,
  registration: string,
  sexo: string,
  email: string,
  password: string,
) {
  try {
    const request = await api.post('/users/signup', {
      name,
      registration,
      sexo,
      email,
      password,
    });

    return request.data;
  } catch {
    return null;
  }
}
