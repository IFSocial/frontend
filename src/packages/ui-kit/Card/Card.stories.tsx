import React from 'react';

import { Story, Meta } from '@storybook/react';

import Card, { CardProps } from './Card';

export default {
  title: 'UI/Card',
  component: Card,
} as Meta;

const Template: Story<CardProps> = function Component(args) {
  return <Card {...args} />;
};

export const Modalidade = Template.bind({});
Modalidade.args = {
  title: 'Nome do Esporte',
  subtitle: 'Inscrever-se',
  image:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b0/Futsal_pictogram.svg/1024px-Futsal_pictogram.svg.png',
};

export const Tipo = Template.bind({});
Tipo.args = {
  subtitle: 'Esportes Coletivos',
  image:
    'https://quizizz.com/media/resource/gs/quizizz-media/quizzes/e4f18483-db85-4a3d-a350-a0c581c4ace4',
};
