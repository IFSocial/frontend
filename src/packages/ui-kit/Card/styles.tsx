import styled from 'styled-components';

export type StyledCardProps = {
  size?: 'small' | 'default';
};

export const Card = styled.div`
  width: 212px;
  height: 265px;
  background: ${({ theme }) => theme.colors.Primary700};
  border-radius: 10px;
  border: 1px solid ${({ theme }) => theme.colors.dark900};
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25),
    inset 0px 4px 4px rgba(0, 0, 0, 0.25);
  cursor: pointer;
  display: flex;
  justify-content: center;
  text-align: center;
  align-items: center;
  flex-direction: column;
  padding: 8px;
`;

export const Title = styled.p`
  font-family: 'Montserrat';
  font-weight: 500;
  font-size: 25px;
  display: flex;
  align-items: center;
  text-align: center;
`;

export const Subtitle = styled.p`
  font-family: 'Montserrat';
  font-weight: 700;
  font-size: 25px;
  color: ${({ theme }) => theme.colors.Primary900};
`;

export const ImageWrapper = styled.div`
  display: flex;
  justify-content: center;
`;

export const Image = styled.img`
  width: 70%;
  height: auto !important;
  object-fit: scale-down;
`;
