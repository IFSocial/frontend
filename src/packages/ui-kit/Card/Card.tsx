import React from 'react';

import * as Styled from './styles';

export interface CardProps extends Styled.StyledCardProps {
  title?: string;
  subtitle: string;
  image: string;
  onClick: () => void;
}

function Card({ title, subtitle, image, onClick }: CardProps) {
  return (
    <Styled.Card onClick={onClick} data-testid="Card">
      <Styled.Title>{title}</Styled.Title>
      <Styled.ImageWrapper>
        <Styled.Image
          data-testid="card-image"
          src={image}
          alt="Imagem do Card"
        />
      </Styled.ImageWrapper>
      <Styled.Subtitle>{subtitle}</Styled.Subtitle>
    </Styled.Card>
  );
}

export default Card;
