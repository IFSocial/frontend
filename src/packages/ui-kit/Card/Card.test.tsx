import React from 'react';

import { fireEvent, render } from '../../test/utils';
import Card from './Card';

describe('Card', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(
      <Card subtitle="" image="" onClick={jest.fn()} />,
    );

    const CardComponent = getByTestId('Card');

    expect(CardComponent).toBeInTheDocument();

    expect(CardComponent).toMatchSnapshot();
  });
  it('should be clickable', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Card subtitle="" image="" onClick={handleClick} />,
    );

    const CardComponent = getByTestId('Card');

    fireEvent.click(CardComponent);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
  it('should have title, subtitle and image', () => {
    const { getByText, getByTestId } = render(
      <Card
        title="Title"
        subtitle="Subtitle"
        image="Image"
        onClick={jest.fn()}
      />,
    );

    const Title = getByText('Title');
    const Subtitle = getByText('Subtitle');
    const Image = getByTestId('card-image');

    expect(Title).toBeInTheDocument();
    expect(Subtitle).toBeInTheDocument();
    expect(Image).toBeInTheDocument();
  });
  it('should have subtitle and image only', () => {
    const { getByText, getByTestId, queryByTestId } = render(
      <Card subtitle="Subtitle" image="Image" onClick={jest.fn()} />,
    );

    const Title = queryByTestId('card-title');
    const Subtitle = getByText('Subtitle');
    const Image = getByTestId('card-image');

    expect(Title).not.toBeInTheDocument();
    expect(Subtitle).toBeInTheDocument();
    expect(Image).toBeInTheDocument();
  });
});
