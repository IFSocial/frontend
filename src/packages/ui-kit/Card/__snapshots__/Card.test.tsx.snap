// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  class="sc-bcXHqe iWcaxZ"
  data-testid="Card"
>
  <p
    class="sc-gswNZR jThlhX"
  />
  <div
    class="sc-hLBbgP gglZrP"
  >
    <img
      alt="Imagem do Card"
      class="sc-eDvSVe bFKOJD"
      data-testid="card-image"
      src=""
    />
  </div>
  <p
    class="sc-dkrFOg eVsFVa"
  />
</div>
`;
